scribejava-core/src/main/java/com/github/scribejava/core/oauth2/OAuth2Error.java

```diff
@@ -67,7 +72,29 @@
      * @see <a href="https://tools.ietf.org/html/rfc7009#section-4.1">RFC 7009, 4.1. OAuth Extensions Error
      * Registration</a>
      */
-    UNSUPPORTED_TOKEN_TYPE("unsupported_token_type");
+    UNSUPPORTED_TOKEN_TYPE("unsupported_token_type"),
+
+    /**
+     * @see <a href="https://tools.ietf.org/html/rfc8628#section-3.5">rfc8628#section-3.5</a>
+     */
+    AUTHORIZATION_PENDING("authorization_pending"),
+
+    /**
+     * @see <a href="https://tools.ietf.org/html/rfc8628#section-3.5">rfc8628#section-3.5</a>
+     */
+    SLOW_DOWN("slow_down"),
+
+    /**
+     * @see <a href="https://tools.ietf.org/html/rfc8628#section-3.5">rfc8628#section-3.5</a>
+     */
+    EXPIRED_TOKEN("expired_token"),
+    ;
+
+    /**
+     * Unlike {@link #values()} which creates a new array every time, this always
+     * returns the same immutable object.
+     */
+    public static final Set<OAuth2Error> VALUES = unmodifiableSet(EnumSet.allOf(OAuth2Error.class));
```

Comment:
...but a bit more memory consumption. CPU vs memory ;-)
ok, I will leave your variant.
except for unmodifiableSet. unneeded wrapper, it's better to make this member private (I have done this)

Reply:
The old variant actually uses more memory and more CPU and copy and GC the array every time values() is called. The wrapper is needed when the field is public, which IMHO should be to let others use it. The wrapper has zero impact on performance.

Reply:
I mean, old variant uses memory only when it's invoked, but in cold state it doesn't need memory at all.

As for public members, I think it can break encapsulation. In case someone will need it outside, we will return to this discussion.

p.s. And for zero impact, I can't agree, at least at application startup, but this impact is soo small, so it's not the subject to discuss ;-)

Reply:
How an immutable field can break encapsulation?

Reply:
*public* member breaks encapsulation, not immutable field

Reply:
Could you please tell me what you mean by "it breaks encapsulation"? I would really like to learn about how to improve my code.

Reply:
you can use google ;-) for example first link
https://www.javatpoint.com/encapsulation

it's a good practice to not have a public member, but to give access to them via public methods. So you can hide real implementations inside the class and change it whenever you need.

Reply:
I understand encapsulation in the context of an object. I totally agree it is very good to have API that prevents programmers from changing internal state of an object freely, possibly making the state invalid. However this field is not part of any object state, it is a constant and I do not think it is a good or common practice to access constants using methods.

Reply:
Yes, you are right, there is no need for access methods for constants, but... to be honest, this member is not a constant by the sense. It's a constant because of modifiers "public static final". Actually this "constant" is a just internal optimisation.
Anyway, I think this discussion is more theoretical and about coding styles. Let's not spend our time on that.
In case of any practical problems with this code, we will return to this question.